src/DebugOnly/DebugOnly.jl

```diff
@@ -98,4 +98,77 @@ function print_depth_limited_stack_trace(
     end
 end
 
+
+"""
+    allow_mismatched_spaces()
+
+When returning `true`, disable check for consistency among spaces in broadcasted
+operations.
+
+By default, `ClimaCore` checks that broadcasted in-place expressions use
+consistent spaces (ie, the destination space is the same as the space that the
+expression returns). Sometimes, when debugging, it is convenient to disable this
+check.
+
+The most common case for this is to allow combining spaces that were
+`deepcopied`, given that the consistency check in performed by comparing the
+pointer of the spaces, not their contents. In other words, allowing for
+mismatched spaces allows one to work with spaces that are identical, but not the
+same.
+
+To allow combining mismatched spaces, override this function so that it returns
+`true`.
+
+!!! warn
+
+    `ClimaCore` checks for consistency of spaces to protect you from non-sense
+    results. If you disable this check, you are responsible to ensure that the
+    results make sense.
+
+Example
+=======
+
+```julia
+julia> import ClimaCore;
+
+julia> using ClimaCore.CommonSpaces;
+
+julia> space = ExtrudedCubedSphereSpace(;
+           z_elem = 10,
+           z_min = 0,
+           z_max = 1,
+           radius = 10,
+           h_elem = 10,
+           n_quad_points = 4,
+           staggering = CellCenter()
+       );
+
+julia> other_space = deepcopy(space);
+
+julia> other_space == space
+false
+
+julia> one = ones(space);
+
+julia> other_one = ones(other_space);
+
+julia> one .+ other_one
+ERROR: Broacasted spaces are not the same.
+Stacktrace:
+ [1] error(s::String)
+   @ Base ./error.jl:35
+ [2] error_mismatched_spaces(space1::Type, space2::Type)
+   @ ClimaCore.Fields ~/repos/ClimaCore.jl/src/Fields/broadcast.jl:227
+
+# Turning `allow_mismatched_spaces` on
+julia> ClimaCore.DebugOnly.allow_mismatched_spaces() = true;
+
+julia> one .+ other_one
+Float64-valued Field:
+  [2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0  …  2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 2.0]
+"""
+function allow_mismatched_spaces()
```

Comment:
If we're going to make this method user-facing, and being that this method opens the door for incorrect use, I'd suggest that we use the word `unsafe`, or similar, somewhere in the name

Reply:
Okay